src/main/java/com/github/shyiko/mysql/binlog/BinaryLogClient.java

```diff
@@ -779,41 +782,52 @@ private void listenForEventPackets() throws IOException {
         } while (chunkLength == Packet.MAX_LENGTH);
         return result;
     }
-
-    private void updatePosition(Event event) {
+    
+    private boolean updateGtid(Event event) {
         EventHeader eventHeader = event.getHeader();
         EventType eventType = eventHeader.getEventType();
-        if (gtidSet != null && eventType == EventType.XID) {
-            advanceGTID();
-        } else
-        if (gtidSet != null && eventType == EventType.QUERY) {
-            QueryEventData queryEventData = getInternalEventData(event);
-            String query = queryEventData.getSql();
-            if ("COMMIT".equals(query) || "ROLLBACK".equals(query) ||
-                (previousEvent != null && previousEvent.getHeader().getEventType() == EventType.GTID &&
-                    !"BEGIN".equals(query))) {
+        if (gtidSet != null) {
+            if (eventType == EventType.XID) {
                 advanceGTID();
+                return true;
             }
-        } else
-        if (gtidSet != null && eventType == EventType.GTID) {
-            if (previousGtidEvent != null) {
-                if (advanceGTID()) {
-                    if (logger.isLoggable(Level.WARNING)) {
-                        logger.log(Level.WARNING, "GtidSet wasn't synchronized before GTID. " +
-                            "Please submit a bug report to https://github.com/shyiko/mysql-binlog-connector-java");
+            if (eventType == EventType.QUERY) {
+                QueryEventData queryEventData = getInternalEventData(event);
+                String query = queryEventData.getSql();
+                if ("COMMIT".equals(query) || "ROLLBACK".equals(query) ||
+                        (previousEvent != null && previousEvent.getHeader().getEventType() == EventType.GTID &&
+                                !"BEGIN".equals(query))) {
+                    advanceGTID();
+                }
+                return true;
+            }
+            if (eventType == EventType.GTID) {
+                if (previousGtidEvent != null) {
+                    if (advanceGTID()) {
+                        if (logger.isLoggable(Level.WARNING)) {
+                            logger.log(Level.WARNING, "GtidSet wasn't synchronized before GTID. " +
+                                    "Please submit a bug report to https://github.com/shyiko/mysql-binlog-connector-java");
+                        }
                     }
                 }
+                return true;
             }
-            previousGtidEvent = event;
```

Comment:
The github diff makes this difficult to see, but this is in the `updateGtid(Event)` method. The line does appear in the `updatePosition(Event,boolean)` method [here](https://github.com/shyiko/mysql-binlog-connector-java/pull/125/files#diff-7addc4e3eed6e9254e088abc015ac8adR837).